web/src/components/shared/NavBar.tsx

```diff
@@ -2,15 +2,38 @@ import React, { PureComponent } from "react";
 import PropTypes from "prop-types";
 import classNames from "classnames";
 import { withRouter } from "@src/utilities/react-router-utilities";
+import { RouteComponentProps } from "react-router-dom";
 import { Link } from "react-router-dom";
 import { Utilities } from "@src/utilities/utilities";
 import ErrorModal from "../modals/ErrorModal";
 import NavBarDropdown from "./NavBarDropdown";
 
 import "@src/scss/components/shared/NavBar.scss";
+import { App } from "@types";
 
-export class NavBar extends PureComponent {
-  constructor(props) {
+interface Props extends Partial<RouteComponentProps> {
```

Comment:
using `Partial` allows me to typecheck the properties used in this component without requiring them when the component is used. `withRouter()` adds these props.